File: src/test/java/com/mateuslh/calc/CalculadoraTest.java
package com.mateuslh.calc;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

@DisplayName("Calculadora")
class CalculadoraTest {

    private final Calculadora calculadora = new Calculadora();

    @Test
    @DisplayName("soma dois valores")
    void deveSomar() {
        assertEquals(7, calculadora.somar(3, 4));
        assertEquals(-1, calculadora.somar(2, -3));
    }

    @Test
    @DisplayName("subtrai dois valores")
    void deveSubtrair() {
        assertEquals(1, calculadora.subtrair(4, 3));
        assertEquals(5, calculadora.subtrair(2, -3));
    }

    @Test
    @DisplayName("multiplica dois valores")
    void deveMultiplicar() {
        assertEquals(12, calculadora.multiplicar(3, 4));
        assertEquals(0, calculadora.multiplicar(0, 7));
    }

    @Test
    @DisplayName("divide dois valores")
    void deveDividir() {
        assertEquals(2, calculadora.dividir(8, 4));
        assertEquals(3, calculadora.dividir(9, 3));
    }

    @Test
    @DisplayName("recusa divisão por zero")
    void deveRecusarDivisaoPorZero() {
        IllegalArgumentException erro = assertThrows(
                IllegalArgumentException.class,
                () -> calculadora.dividir(10, 0));
        assertEquals("Divisão por zero não é permitida", erro.getMessage());
    }

    @Test
    @DisplayName("identifica números pares e ímpares")
    void deveIdentificarPar() {
        assertTrue(calculadora.ehPar(4));
        assertTrue(calculadora.ehPar(0));
        assertFalse(calculadora.ehPar(3));
        assertFalse(calculadora.ehPar(-7));
    }

    @Test
    @DisplayName("retorna o maior valor")
    void deveRetornarMaximo() {
        assertEquals(5, calculadora.maximo(5, 3));
        assertEquals(5, calculadora.maximo(3, 5));
        assertEquals(5, calculadora.maximo(5, 5));
    }
}

File: src/calculadora.test.ts
import { describe, expect, it } from 'vitest';
import { Calculadora } from './calculadora.js';

describe('Calculadora', () => {
  const calculadora = new Calculadora();

  it('soma dois valores', () => {
    expect(calculadora.somar(3, 4)).toBe(7);
    expect(calculadora.somar(2, -3)).toBe(-1);
  });

  it('subtrai dois valores', () => {
    expect(calculadora.subtrair(4, 3)).toBe(1);
    expect(calculadora.subtrair(2, -3)).toBe(5);
  });

  it('multiplica dois valores', () => {
    expect(calculadora.multiplicar(3, 4)).toBe(12);
    expect(calculadora.multiplicar(0, 7)).toBe(0);
  });

  it('divide dois valores', () => {
    expect(calculadora.dividir(8, 4)).toBe(2);
    expect(calculadora.dividir(9, 3)).toBe(3);
  });

  it('recusa divisão por zero', () => {
    expect(() => calculadora.dividir(10, 0)).toThrowError(
      'Divisão por zero não é permitida',
    );
  });

  it('identifica números pares e ímpares', () => {
    expect(calculadora.ehPar(4)).toBe(true);
    expect(calculadora.ehPar(0)).toBe(true);
    expect(calculadora.ehPar(3)).toBe(false);
    expect(calculadora.ehPar(-7)).toBe(false);
  });

  it('retorna o maior valor', () => {
    expect(calculadora.maximo(5, 3)).toBe(5);
    expect(calculadora.maximo(3, 5)).toBe(5);
    expect(calculadora.maximo(5, 5)).toBe(5);
  });
});
